test/test-parse-loop.cpp

```diff
@@ -893,7 +893,8 @@ TEST_F(Test_Parse_Loop, for_of_loop) {
 
   {
     Spy_Visitor p = test_parse_and_visit_statement(
-        u8"for await (let x of xs) { body; }"_sv, no_diags, javascript_options);
+        u8"for await (let x of xs) { body; }"_sv,  //
+        u8"    ^^^^^ Diag_Await_Operator_Outside_Async"_diag);
```

Comment:
Must fix: This is wrong, actually. `await` is allowed at the top level.

There are five cases:
1. top level with Parser_Top_Level_Await_Mode::auto_detect
2. top level with Parser_Top_Level_Await_Mode::await_operator
3. top level with Parser_Top_Level_Await_Mode::no_await_operator (not implemented in quick-lint-js yet)
4. in non-async function
5. in async function

Cases 1, 2, and 5 should emit no diagnostic. Cases 3 and 4 should emit a diagnostic. This line is case 1, so there should be no diagnostic.

Reply:
That makes sense- thanks for outlining all the cases! I've updated my commit to revert the change I made to this test, and added a check to make sure that we're not at the top level before reporting the diagnostic.

Since #1137 is regarding an error on `for await...of` in a non-async function, rather than at the top-level, should we leave the third case as a TODO?

Reply:
> should we leave the third case as a TODO?

Yeah, your TODO comment is fine.